src/main/java/org/opensearch/neuralsearch/processor/combination/ArithmeticMeanScoreCombinationTechnique.java

```diff
@@ -52,7 +38,7 @@ public float combine(final float[] scores) {
         for (int indexOfSubQuery = 0; indexOfSubQuery < scores.length; indexOfSubQuery++) {
             float score = scores[indexOfSubQuery];
             if (score >= 0.0) {
-                float weight = getWeightForSubQuery(indexOfSubQuery);
+                float weight = getWeightForSubQuery(this.weights, indexOfSubQuery);
```

Comment:
Can we rename local variable name to a different one to avoid future mistake?

Reply:
ack